From fe4352ca97653db55892c41a80f505018ad5a8b5 Mon Sep 17 00:00:00 2001
From: James Page <james.page@chainguard.dev>
Date: Fri, 15 Aug 2025 12:43:45 +0100
Subject: [PATCH 2/2] Add missing entrypoints for provider management functions

Add entrypoints for CryptoNative_LoadDefaultProvider,
CryptoNative_UnloadProvider, and CryptoNative_CleanupDefaultProvider
to fix build errors.
---
 .../libs/System.Security.Cryptography.Native/entrypoints.c     | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/runtime/src/native/libs/System.Security.Cryptography.Native/entrypoints.c b/src/runtime/src/native/libs/System.Security.Cryptography.Native/entrypoints.c
index de513a3c415..ba535a8c299 100644
--- a/src/runtime/src/native/libs/System.Security.Cryptography.Native/entrypoints.c
+++ b/src/runtime/src/native/libs/System.Security.Cryptography.Native/entrypoints.c
@@ -51,6 +51,7 @@ static const Entry s_cryptoNative[] =
     DllImportEntry(CryptoNative_BioWrite)
     DllImportEntry(CryptoNative_CheckX509Hostname)
     DllImportEntry(CryptoNative_CheckX509IpAddress)
+    DllImportEntry(CryptoNative_CleanupDefaultProvider)
     DllImportEntry(CryptoNative_CreateMemoryBio)
     DllImportEntry(CryptoNative_D2IPkcs7Bio)
     DllImportEntry(CryptoNative_DecodeAsn1BitString)
@@ -236,6 +237,7 @@ static const Entry s_cryptoNative[] =
     DllImportEntry(CryptoNative_LoadKeyFromProvider)
     DllImportEntry(CryptoNative_LoadPrivateKeyFromEngine)
     DllImportEntry(CryptoNative_LoadPublicKeyFromEngine)
+    DllImportEntry(CryptoNative_LoadDefaultProvider)
     DllImportEntry(CryptoNative_LookupFriendlyNameByOid)
     DllImportEntry(CryptoNative_NewX509Stack)
     DllImportEntry(CryptoNative_ObjNid2Obj)
@@ -262,6 +264,7 @@ static const Entry s_cryptoNative[] =
     DllImportEntry(CryptoNative_RsaGenerateKey)
     DllImportEntry(CryptoNative_RsaSignHash)
     DllImportEntry(CryptoNative_RsaVerifyHash)
+    DllImportEntry(CryptoNative_UnloadProvider)
     DllImportEntry(CryptoNative_UpRefEvpPkey)
     DllImportEntry(CryptoNative_EvpPKeyType)
     DllImportEntry(CryptoNative_X509BuildOcspRequest)
-- 
2.43.0

